src/api/hash.rs

```diff
@@ -2,9 +2,10 @@
 
 macro_rules! impl_hash {
     ([$elem_ty:ident; $elem_count:expr]: $id:ident | $test_tt:tt) => {
-        impl ::hash::Hash for $id {
+        impl core::hash::Hash for $id {
```

Comment:
We cannot use `core::` or `std::` in the main library because that won't work when we include it in `std` or `core` - it also makes it a pain to update the library if we ever want to switch to requiring `std` because of float-math support. 

We should use the imports at the top level in `lib.rs`, and the rest of the code should only use relative imports. This makes adding a shim to directly include the library in `libcore` trivial.